sw/host/opentitanlib/src/rescue/serial.rs

```diff
@@ -36,23 +41,52 @@ impl RescueSerial {
             uart,
             reset_delay: Duration::from_millis(50),
             enter_delay: Duration::from_secs(5),
+            version: Cell::default(),
         }
     }
 }
 
 impl Rescue for RescueSerial {
-    fn enter(&self, transport: &TransportWrapper, reset_target: bool) -> Result<()> {
-        log::info!("Setting serial break to trigger rescue mode.");
-        self.uart.set_break(true)?;
-        if reset_target {
-            transport.reset_target(self.reset_delay, /*clear_uart=*/ true)?;
+    fn enter(&self, transport: &TransportWrapper, mode: EntryMode) -> Result<()> {
+        log::info!("Setting serial break to trigger rescue mode (entry via {mode:?}).");
+        match mode {
+            EntryMode::Reset => {
+                self.uart.set_break(true)?;
+                transport.reset_target(self.reset_delay, /*clear_uart=*/ true)?;
+            }
+            EntryMode::Reboot => {
+                self.reboot()?;
+                self.uart.set_break(true)?;
+            }
+            EntryMode::None => {
+                self.uart.set_break(true)?;
```

Comment:
Does the ROM_EXT print its banner again, (and reset to initial mode), if it detects break condition?  That makes sense.

Reply:
Break, on its own, does not affect the rescue state.  `EntryMode::None` is for test cases that may have some out-of-band mechanism of resetting the chip or triggering rescue mode.

We set break condition so that if the chip goes through reset, rescue will be triggered again.